src/Guard.php

```diff
@@ -236,6 +255,28 @@ protected function removeFromStorage($name)
     }
 
     /**
+     * Remove the oldest tokens from the storage array so that there
+     * are never more than storageLimit tokens in the array.
+     *
+     * This is required as a token is generated every request and so
+     * most will never be used.
+     */
+    protected function enforceStorageLimit()
+    {
+        if ($this->storageLimit < 1) {
+            return;
+        }
+
+        if (!is_array($this->storage) && !$this->storage instanceof Countable) {
+            return;
+        }
+
+        while (count($this->storage) > $this->storageLimit) {
+            array_shift($this->storage);
```

Comment:
As session items aren't dated, this call doesn't ensure the removed token is the oldest, does it? Though, I guess a dated token mechanism would be much more complicated.